Extend the documentation of Custom Disk Volumes (#1073)
SUSE/kiwi: doc/source/building/working_with_images/custom_volumes.rst

.. sidebar:: Abstract

   This chapter provides high level explanations on how to handle volumes
   or subvolumes definitions for disk images using KIWI.

KIWI supports defining custom volumes by using the logical volume manager
(LVM) for the Linux kernel or by setting volumes at filesystem level when
filesystem supports it (e.g. btrfs).

Volumes are defined in the KIWI description file :file:`config.xml`,

> using the `systemdisk` element, which itself is a child of the `type`

Reviewer: I would suggest to shorten the paragraph a bit:

```suggestion
using `systemdisk`. This element is a child of the `type`.
```

> element. Volumes themselves are added via (multiple) `volume` child

Reviewer: ```suggestion
Volumes themselves are added via (multiple) `volume` child
```

elements of the `systemdisk` element:

.. code-block:: xml

   <image schemaversion="7.1" name="openSUSE-Leap-15.1">
     <type image="oem" filesystem="btrfs" preferlvm="true">
       <systemdisk name="vgroup">
         <volume name="usr/lib" size="1G" label="library"/>
         <volume name="@root" freespace="500M"/>
         <volume name="etc_volume" mountpoint="etc" copy_on_write="false"/>
         <volume name="bin_volume" size="all" mountpoint="/usr/bin"/>
       </systemdisk>
     </type>
   </image>

> Additional non-root volumes are created for each present `volume`

Reviewer: IHMO, the "present" word is not needed:

```suggestion
Additional non-root volumes are created for each `volume`
```

element. Volume details can be defined by setting the following `volume`
attributes:

> - `name`: Required attribute representing a volume's name. In addition this

Reviewer: ```suggestion
- `name`: Required attribute representing the volume's name. Additionally, this
```

  attribute is interpreted as the mountpoint if the `mountpoint` attribute
  is not used.

- `mountpoint`: Optional attribute that specifies the mountpoint of this
  volume.

- `size`: Optional attribute to set the size of the volume. If no suffix
  (`M` or `G`) is used, then the value is considered to be in megabytes.

  .. note:: Special name for the root volume

     One can use the `@root` name to refer to the volume mounted at `/`, in
     case some specific size attributes for the root volume have to be
     defined. For instance:

     .. code-block:: xml

        <volume name="@root" size="4G"/>

- `freespace`: Optional attribute defining the additional free space added

>   to the volume. If no suffix (`M` or `G`) is used the value is considered

Reviewer: ```suggestion
  to the volume. If no suffix (`M` or `G`) is used, the value is considered
```

  to be in megabytes.

- `label`: Optional attribute to set filesystem label of the volume.

- `copy_on_write`: Optional attribute to set the filesystem copy-on-write
  attribute for this volume.

.. warning::

>    Note that size attributes for filesystem volumes, as for btrfs, are

Reviewer: "Note" inside a warning is superfluous.

```suggestion
   The size attributes for filesystem volumes, as for btrfs, are
```

   ignored and have no effect.

The `systemdisk` element additionally supports the following optional
attributes:

- `name`: The volume group name, by default `kiwiVG` is used. This setting
  is only relevant for LVM volumes.

> - `preferlvm`: Prefer LVM even if the used filesystem has its own volume

Reviewer: Is the value a boolean datatype? Which values are allowed?

Author: It's a boolean, I'll make this clearer.
